resources/RegexGrammar.pp

```diff
@@ -88,7 +90,7 @@
 %token  nc:_named_capturing      >                  -> default
 %token  nc:capturing_name        .+?(?=(?<!\\)>)
 %token  non_capturing_           \(\?:
-%token  non_capturing_internal_option \(\?([imsxnJUX^]|xx)?-?([imsxnJUX^]|xx):
+%token  non_capturing_internal_option \(\?[imsxnJUX]*-?[imsxnJUX]+:
```

Comment:
why did these change?

Reply:
see added testcase - `xx` is not PCRE syntax, in theory we still support is as newly multiple options are allowed (per valid PCRE grammar). `^` - I am not sure what it meant, I think it is not supported by PCRE.

Reply:
Needs the same changes as above in `internal_option` for `^` 

And btw `xx` is also described on https://www.regular-expressions.info/refmodifiers.html - it acts like x but also supports spaces inside character classes. It only works as an inline modifier though, not in main regex modifiers https://3v4l.org/ssAQn

Reply:
But btw no need to support `xx` specially here, as you now allow more than one modifier xx is anyway valid now just like `mmmmmmmm` will be valid too. This is only relevant information when handling modifiers in the PHP code later, where you want to make sure that if there is an internal option with `xx` then the character class handling should handle spaces differently in what follows.

Reply:
done and thank you